charts/trino/templates/networkpolicy.yaml

```
@@ -0,0 +1,38 @@
+{{- if .Values.networkPolicy.enabled }}
+{{- if eq "NodePort" .Values.service.type}}
+{{- fail "NetworkPolicy enforcement is not supported with NodePort Services, as traffic reaches the Pod through the node itself, bypassing Pod-level network controls." }}
```

It's great to have this! Nit: I think names of Kubernetes resources should start with a lower case.
```suggestion
{{- fail "Network policy enforcement is not supported with `NodePort` services, as traffic reaches the pod through the node itself, bypassing pod-level network controls." }}
```

---

Have a look at this: https://kubernetes.io/docs/contribute/style/style-guide/#use-upper-camel-case-for-api-objects

---

I was thinking about this example, where both a resource name (`PodList`)` and a noun (`pods`) are used.
> A PodList object is a list of pods.

I've edited my suggestion to only revert to `NetworkPolicy`. Thanks for the reference, it'll be very useful in the future!

---

I think `Pod` here should also be capitalized.

---

**NetworkPolicy** enforcement is not supported with **NodePort** services, as traffic reaches the **Pod** through the node itself, bypassing pod-level network controls.